### PR TITLE
Add the `USER_CAN_ACCESS_PAGE` permission

### LINKED ISSUE #4818: Inconsistent deprecations
While working on #4816, I have noticed that the deprecations are inconsistent:

**We have deprecated**

* `BackendUser::isAllowed()`
* `BackendUser::canEditFieldsOf()`
* `FrontendUser::isMemberOf()`

**But we have not deprecated**

* `BackendUser::hasAccess()`
* `User::isMemberOf()`

@contao/developers Why is that?

_Originally posted by @leofeyer in https://github.com/contao/contao/issues/4816#issuecomment-1152109114_

### PR DESCRIPTION
**Pull request by aschempp**
see https://github.com/contao/contao/issues/4818

### CODE REVIEW

File: core-bundle/contao/classes/BackendUser.php [hunk 1]
```diff
@@ -155,6 +155,8 @@ public function __get($strKey)
 	 */
 	public function hasAccess($field, $array)
 	{
+		trigger_deprecation('contao/core-bundle', '5.2', 'Using "%s()" has been deprecated and will no longer work in Contao 6. Use the "ContaoCorePermissions::USER_CAN_ACCESS_*" permissions instead.', __METHOD__);
+
 		if ($this->isAdmin)
```
Comment (Toflar) [inline core-bundle/contao/classes/BackendUser.php:160]: Should we forward those to the new voter? Or is it too complicated?

Comment (aschempp) [reply core-bundle/contao/classes/BackendUser.php:160]: I don't think so. We don't do that on the other methods, and technically you could use this method on a User object that is not the current Symfony security user.

**Review by Toflar — DISMISSED** [review]
LGTM, CS fixer missing though.

**Review by Toflar — DISMISSED** [review]

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Pull request merged**